hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JobSerializerTest.java

```diff
@@ -0,0 +1,207 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core;
+
+import com.hazelcast.cache.ICache;
+import com.hazelcast.config.CacheSimpleConfig;
+import com.hazelcast.config.SerializerConfig;
+import com.hazelcast.jet.Observable;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.Sources;
+import com.hazelcast.jet.pipeline.test.AssertionSinks;
+import com.hazelcast.jet.pipeline.test.TestSources;
+import com.hazelcast.map.IMap;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.StreamSerializer;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableSet;
+
+import javax.cache.Cache;
+import java.io.IOException;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class JobSerializerTest extends SimpleTestInClusterSupport {
+
+    private static final String SOURCE_MAP_NAME = "source-map";
+    private static final String SINK_MAP_NAME = "sink-map";
+    private static final String SOURCE_CACHE_NAME = "source-cache";
+    private static final String SINK_CACHE_NAME = "sink-cache";
+    private static final String OBSERVABLE_NAME = "observable";
+
+    @BeforeClass
+    public static void beforeClass() {
+        JetConfig config = new JetConfig();
+        config.getHazelcastConfig()
+              .addCacheConfig(new CacheSimpleConfig().setName(SOURCE_CACHE_NAME))
+              .addCacheConfig(new CacheSimpleConfig().setName(SINK_CACHE_NAME))
+              .getSerializationConfig()
+              .addSerializerConfig(new SerializerConfig().setTypeClass(Value.class).setClass(ValueSerializer.class));
```

Comment:
We should not register the serializer for the cluster: the test is supposed to test that even if it's not set for the cluster, it works if it's set for the job.